p2panda-rs/src/secret_group/commit.rs

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use openmls::framing::{
+    MlsMessageIn, MlsMessageOut, MlsPlaintextContentType, VerifiableMlsPlaintext,
+};
+use openmls::messages::Welcome;
+use tls_codec::{Size, TlsDeserialize, TlsSerialize, TlsSize};
+
+use crate::secret_group::{SecretGroupError, SecretGroupMessage};
+
+/// Plaintext commit message which is published on the network to announce group changes.
+///
+/// A `SecretGroupCommit` always contains a MLS Commit message and the current, encrypted long term
+/// secrets for this group. Optionally it contains another MLS Welcome message in case this commit
+/// invites new members into the group.
+#[derive(Debug, Clone, TlsSerialize, TlsDeserialize, TlsSize)]
+pub struct SecretGroupCommit {
+    mls_commit_message: VerifiableMlsPlaintext,
```

Comment:
You shouldn't have to deal with `VerifiableMlsPlaintext` directly and use `MlsMessageIn`/`MlsMessageOut` instead. If that doesn't work for some reason, feel free to file an issue.

Reply:
`MlsMessageOut` does not implement the `TlsDeserialize` trait which makes sense but for p2panda we don't really have a notion of a "delivery service" or at least one which does not know about MLS specifically.

Reply:
I filed https://github.com/openmls/openmls/issues/585 for this. 

My point was that you do have a DS, even if it is client-side. It's probably best to treat messages as outgoing messages (`MlsMessageOut`) until they reach a point where the intention is to process them as incoming messages. That's when you could convert them to `MlsMessageIn`.

Reply:
> My point was that you do have a DS, even if it is client-side. It's probably best to treat messages as outgoing messages (MlsMessageOut) until they reach a point where the intention is to process them as incoming messages. That's when you could convert them to MlsMessageIn.

Yes, we're doing this here:

https://github.com/p2panda/p2panda/blob/mls-core/p2panda-rs/src/secret_group/commit.rs#L53-L56